### ISSUE TITLE
Incorrect local variable scope in CalcInteriorSolarDistribution

### ISSUE THREAD

**Issue by xuanluo113** [NREL/EnergyPlus#8369]
A potential bug in solar calculations in CalcInteriorSolarDistribution.

The reporting at the end of the function starts a new surface loop, but the [WinTransBmBmSolar](https://github.com/NREL/EnergyPlus/blob/649ab487ef39cf44c2a8703beaeb750d25e687bd/src/EnergyPlus/SolarShading.cc#L6206), WinTransBmDifSolar, [TBmBm + TBmDif](https://github.com/NREL/EnergyPlus/blob/649ab487ef39cf44c2a8703beaeb750d25e687bd/src/EnergyPlus/SolarShading.cc#L5998) are not re-calculated. The value, [WinTransBmBmSolar](https://github.com/NREL/EnergyPlus/blob/649ab487ef39cf44c2a8703beaeb750d25e687bd/src/EnergyPlus/SolarShading.cc#L8254), WinTransBmBmSolar, and [TBmBm + TBmDif](https://github.com/NREL/EnergyPlus/blob/649ab487ef39cf44c2a8703beaeb750d25e687bd/src/EnergyPlus/SolarShading.cc#L8260) used in the reporting loop are always the last calculated value for the last surface in the previous loop.

Defect file: testfiles/PurchAirWindowBlind_BlockBeamSolar.idf

### Details
Some additional details for this issue (if relevant):
 - Platform (Operating system, version)
 - Version of EnergyPlus (if using an intermediate build, include SHA)
 - Unmethours link or helpdesk ticket number

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here)
 - [ ] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

**Comment by mjwitte** [comment]
@xuanluo113 If I am following this correctly, this only impacts the following output variables. It does not affect the actual heat balance of the zone.
```
Window Transmitted Beam-to-Beam Solar
Window Transmitted Beam-to-Diffuse Solar
Window Transmitted Beam-to-Beam Solar Energy
Window Transmitted Beam-to-Diffuse Solar Energy
```
And these variables have been wrong since they were first added in 2009/2010.

**Comment by xuanluo113** [comment]
@mjwitte I agree. 
1. The only reportable variables affected are:
- Surface Window Transmitted Beam To Beam Solar Radiation Rate
- Surface Window Transmitted Beam To Diffuse Solar Radiation Rate
- Surface Window Transmitted Beam To Beam Solar Radiation Energy
- Surface Window Transmitted Beam To Diffuse Solar Radiation Energy

This may only affect the result of `PurchAirWindowBlind_BlockBeamSolar.idf` that currently reports those. However, there is only one window per zone in this model, so using calculations of the last (and only) window in the reporting loop does not affect anything, coincidentally. 

2. The other affected variable `SurfWinDirSolTransAtIncAngle(SurfNum)// For TDD:DIFFUSER this is the TDD transmittance` are not actually used anywhere, since the transmittance for the diffuser surface should be calculated as windows. My understanding is the variable declaration and calculation could be deleted. Would you mind confirming this by taking another look?

The fixes in PR #8358 show no diff.
